lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -241,10 +241,17 @@ MemBehavior MemoryBehaviorVisitor::visitApplyInst(ApplyInst *AI) {
   MemBehavior Behavior = MemBehavior::None;
 
   // We can ignore mayTrap().
-  if (ApplyEffects.mayReadRC() ||
-      (InspectionMode == RetainObserveKind::ObserveRetains &&
-       ApplyEffects.mayAllocObjects())) {
-    Behavior = MemBehavior::MayHaveSideEffects;
+  if (any_of(enumerate(AI->getArgumentOperands()),
```

Comment:
@eeckstein 

 - `op.Index = 1`
 - `AI->...->getNumParams` = 1
 - `V` =   `%1 = argument of bb0 : $*AnyIterator<τ_0_0>    // user: %3`
 - `*AI` =  `%8 = apply %7<AnyIterator<τ_0_0>>(%0, %2) : $@convention(method) <τ_0_0 where τ_0_0 : Sequence, τ_0_0 == τ_0_0.Iterator> (@in_guaranteed τ_0_0) -> @out τ_0_0`

is it maybe the `@out` getting its storage passed as an argument?